aya_tamimi/task_endpoints.py

```diff
@@ -0,0 +1,97 @@
+from fastapi import FastAPI, HTTPException
+import uuid
+from models import *
+
+
+app = FastAPI()
+tasks = []
+
+# Create a new task with details like title, description, due date, and status
+@app.post("/tasks/")
+def create_task(task :Create_Task_Basemodel):
+    try:
+        if task.priority:
+            new_task = UrgentTask(task.priority , task.title,task.description ,task.due_date,task.status)
+        else :
+           new_task = Task(task.title,task.description ,task.due_date,task.status)
+        tasks.append(new_task)
+        return new_task.__dict__
+    except Exception as e:
+            return {"message": f"An error occurred while creating the task :  {str(e)}"}
+
+
+# Update the details of an existing task by its ID.
+@app.put("/tasks/{task_id}")
+def update_task(task_id: uuid.UUID, task_update: Update_Task_Basemodel):
+    try:
+        task_to_update = find_task_by_id(task_id)
+        if not task_to_update:
+            raise HTTPException(status_code=404, detail="Task not found")
+        # Update the task attributes if they are provided
```

Comment:
What happens if the attributes are more? 10? 20? 
are u sure to use this way? 
let us discuss the problem: you do the same operations for all properties for an object, I think it is better to use a loop to iterate over all of them and handle these operations.

Reply:
I’ve switched to a loop for dynamic updates to keep things scalable. 
Appreciate the suggestion made a big difference!